### PR TITLE
[Network] ip and host connectivity checks

### PR DESCRIPTION
**Pull request by mazinamdar**
Signed-off-by: mazinamdar <mazhar.inamdar@seagate.com>

 1. network ip_connectivity <ip1> <ip2> <ip3>
 2. network host_connectivity <host1> <host2>

### CODE REVIEW

File: py-utils/src/utils/validator/v_network.py [hunk 1]
```diff
@@ -68,6 +71,28 @@ def validate_ip_connectivity(self, ips):
             raise VError(
                 errno.ECONNREFUSED, "Ping failed for IP(s). %s" % unreachable_ips)
 
+    def validate_host_connectivity(self, hosts):
+        """Check if hosts are reachable."""
+
+        unreachable_hosts = []
+        for host in hosts:
```
Comment (ujjwalpl) [inline py-utils/src/utils/validator/v_network.py:78]: We can have one interface i.e. validate_host_connectivity for both host and ip so they can be mixed up. More flexible. 

$ validate network connectivity host1.domain1.com 8.8.8.8 host2 ip2

Logic can be

```
for host in hosts:
    ip = host # assume host is ip for the time being. 
    if not self._is_ip(ip): 
        # resolve host from host to ip
        ip = self._resolve_host(host)
    self._validate_ip_connectivity(ip) 

def _validate_ip_connectivity(ip):
....
```

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

File: py-utils/src/utils/validator/v_network.py (hunk 1 above)
Comment (mazinamdar) [reply py-utils/src/utils/validator/v_network.py:73]: comment is addressed

File: py-utils/src/utils/validator/v_network.py [hunk 2]
```diff
@@ -82,8 +90,23 @@ def validate_passwordless_ssh(self, user, nodes):
                        f"CMD {cmd} failed. {run_result[0]}. {run_result[1]}")
                 raise VError(errno.ECONNREFUSED, res)
 
+    def _is_ip(self, ip):
```
Comment (ujjwalpl) [inline py-utils/src/utils/validator/v_network.py:93]: Can be simplified 

```
def _is_ip(self, ip): 
    return if ip.count(".") == 3 and all(self._is_valid_ipv4_part(ip_part) \
           for ip_part in ip.split("."))
```

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

File: py-utils/src/utils/validator/v_network.py [hunk 3]
```diff
@@ -82,8 +90,23 @@ def validate_passwordless_ssh(self, user, nodes):
...
+        if ip.count(".") == 3 and all(self._is_valid_ipv4_part(ip_part)
+           for ip_part in ip.split(".")):
+            is_ip = True
```
[the remark this answers is not in the document]
Comment (mazinamdar) [reply py-utils/src/utils/validator/v_network.py]: actually this was not indentation error,for loop is belong to if condition ,BTW I have removed this added simplified version

File: py-utils/src/utils/validator/v_network.py (hunk 2 above)
Comment (mazinamdar) [reply py-utils/src/utils/validator/v_network.py:93]: comment is addressed

**Review by ujjwalpl — APPROVED** [review]

**Review by ajaysrivas — APPROVED** [review]

**Pull request merged**